src/server/object_services/schemas/object_md_schema.js

```diff
@@ -68,6 +68,10 @@ module.exports = {
         // NOTE: only updated once upload ends
         num_parts: { type: 'integer' },
 
+        // number of objects S3 multiparts for the object created by multipart upload
+        // NOTE: only updated once the upload of all parts is completed
+        num_multiparts: { type: 'integer' },
```

Comment:
What about upgrading for pre-existing objects?
I understand that you have a lazy update in ```calculate_num_multiparts``` upon reads, but this is a weird state in the database (hence your NOTE comment is invalid for pre-existing objects until they would have been read once after upgrade)

Reply:
in general, we don't want to upgrade object-mds, mainly for scale reasons. I think that the lazy update is reasonable